internal/pkg/task/config_runner.go

```diff
@@ -104,3 +107,65 @@ func (r *ConfigRunner) validateDependencies() error {
 
 	return nil
 }
+
+// CheckNonZeroExitCode returns the status of the containers part of the given tasks.
+func (r *ConfigRunner) CheckNonZeroExitCode(tasks []*Task) error {
+	var essentialContainers []string
```

Comment:
i think this could be simpler if this was `map[string]bool` instead (maybe like `isEssential`?)